File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/MallpayPrice.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class MallpayPrice : ApiBase
{
    [JsonProperty("amount")] public long Amount { get; set; }


    [JsonProperty("currency")] public string? Currency { get; set; }


    public string? ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, Amount);
        Add(sb, Currency);
        DeleteLast(sb);
        return sb.ToString();
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/MallpayPrice.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@EqualsAndHashCode(callSuper = false)
@JsonInclude(value = JsonInclude.Include.NON_NULL)
@AllArgsConstructor @NoArgsConstructor
public class MallpayPrice  extends ApiBase implements Signable {
    private Long amount;
    private String currency;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, amount);
        add(sb, currency);
        deleteLast(sb);
        return sb.toString();
    }
}